dvc/cli.py

```diff
@@ -116,11 +151,96 @@ def parse_args(self, args=None, namespace=None):
         # NOTE: overriding to provide a more granular help message.
         # E.g. `dvc plots diff --bad-flag` would result in a `dvc plots diff`
         # help message instead of generic `dvc` usage.
-        args, argv = self.parse_known_args(args, namespace)
-        if argv:
-            msg = "unrecognized arguments: %s"
-            self.error(msg % " ".join(argv), getattr(args, "func", None))
-        return args
+        try:
+            args, argv = self.parse_known_args(args, namespace)
+        except argparse.ArgumentError:
+            if args is None:
+                args = sys.argv[1:]
+            else:
+                args = list(args)
+            self._find_cmd_suggestions(args)
```

Comment:
Let's just let the helper decide that it needs to use sys.argv, no need to do this right here.
```suggestion
            self._find_cmd_suggestions(args)
```

Reply:
I can move this if statement inside of `_find_cmd_suggestions` 👍